airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresType.java

```diff
@@ -0,0 +1,133 @@
+package io.airbyte.integrations.source.postgres;
+
+import java.sql.SQLType;
+import java.sql.Types;
+
+public enum PostgresType  implements SQLType {
+
+    BIT(Types.BIT),
+    TINYINT(Types.TINYINT),
+    SMALLINT(Types.SMALLINT),
+    INTEGER(Types.INTEGER),
+    BIGINT(Types.BIGINT),
+    FLOAT(Types.FLOAT),
+    REAL(Types.REAL),
+    DOUBLE(Types.DOUBLE),
+    NUMERIC(Types.NUMERIC),
+    DECIMAL(Types.DECIMAL),
+    CHAR(Types.CHAR),
+    VARCHAR(Types.VARCHAR),
+    LONGVARCHAR(Types.LONGVARCHAR),
+    DATE(Types.DATE),
+    TIME(Types.TIME),
+    TIMESTAMP(Types.TIMESTAMP),
+    BINARY(Types.BINARY),
+    VARBINARY(Types.VARBINARY),
+    LONGVARBINARY(Types.LONGVARBINARY),
+    NULL(Types.NULL),
+    OTHER(Types.OTHER),
+    JAVA_OBJECT(Types.JAVA_OBJECT),
+    DISTINCT(Types.DISTINCT),
+    STRUCT(Types.STRUCT),
+    ARRAY(Types.ARRAY),
+    BLOB(Types.BLOB),
+    CLOB(Types.CLOB),
+    REF(Types.REF),
+    DATALINK(Types.DATALINK),
+    BOOLEAN(Types.BOOLEAN),
+    ROWID(Types.ROWID),
+    NCHAR(Types.NCHAR),
+    NVARCHAR(Types.NVARCHAR),
+    LONGNVARCHAR(Types.LONGNVARCHAR),
+    NCLOB(Types.NCLOB),
+    SQLXML(Types.SQLXML),
+    REF_CURSOR(Types.REF_CURSOR),
+    TIME_WITH_TIMEZONE(Types.TIME_WITH_TIMEZONE),
+    TIMESTAMP_WITH_TIMEZONE(Types.TIMESTAMP_WITH_TIMEZONE),
+    VARCHAR_ARRAY(Types.ARRAY),
```

Comment:
is there any way to avoid these duplicated types? Currently this means that if we later need to add a type, we have to remember to add both `WHATEVER_NEW_TYPE` and `WHATEVER_NEW_TYPE_ARRAY`

also - my understanding is that postgres doesn't support nested arrays, is that correct? I.e. we won't need to support `TEXT_ARRAY_ARRAY`

Reply:
> is there any way to avoid these duplicated types? Currently this means that if we later need to add a type, we have to remember to add both `WHATEVER_NEW_TYPE` and `WHATEVER_NEW_TYPE_ARRAY`
> 
> also - my understanding is that postgres doesn't support nested arrays, is that correct? I.e. we won't need to support `TEXT_ARRAY_ARRAY`

no, we don't need to remember that when adding a WHATEVER_NEW_TYPE, we also need to add an array of this type WHATEVER_NEW_TYPE_ARRAY

Postgres supports a limited number of arrays, you can see them [here](https://github.com/pgjdbc/pgjdbc/blob/b73899106949e92e497e3143b24f613cf2f9f203/pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java#L84
)